modmesh/apputil.py

```diff
@@ -57,46 +58,68 @@ class AppEnvironment:
     :ivar locals:
         The local namespace of the application.
     """
-    def __init__(self, name):
+
+    # default config
+    _config = {
+        'appEnvName': None,
+        'redirectStdOutFile': 'stdout.txt',
+        'redirectStdErrFile': 'stderr.txt',
+    }
+
+    def __init__(self, config=None):
         self.globals = {
             # Give the application an alias of the top package.
             'mm': importlib.import_module('modmesh'),
             'appenv': self,
         }
         self.locals = {}
-        self.name = name
         # Each run of the application appends a new environment.
-        environ[name] = self
+        environ[config['appEnvName']] = self
 
-    def run_code(self, code):
-        exec(code, self.globals, self.locals)
+        # replace the config value; otherwise keep the default value
+        for k, v in config.items():
+            if v is not None:
+                self._config[k] = v
 
-
-def get_appenv(name=None):
-    if None is name:
+    def run_code(self, code):
+        with open(self._config['redirectStdOutFile'], 'w') as f1:
+            with contextlib.redirect_stdout(f1):
+                with open(self._config['redirectStdErrFile'], 'w') as f2:
+                    with contextlib.redirect_stderr(f2):
+                        try:
+                            exec(code, self.globals, self.locals)
+                        except Exception as e:
+                            print(("{}: {}".format(
+                                type(e).__name__, str(e))), file=sys.stderr)
+                            print("traceback:", file=sys.stderr)
+                            traceback.print_stack()
+                        sys.stdout.flush()
+                        sys.stderr.flush()
+
+
+def new_appenv(config=None):
+    name = config.get('appEnvName', None)
+    if None is name or name == '':
         i = 0
         name = 'anonymous%d' % i
         while name in environ:
             i += 1
             name = 'anonymous%d' % i
```

Comment:
Is there an upper limit to this?  Maybe it would be better to rewrite it as 
```suggestion
        for i in range(10):
            name = f'anonymous{i}'
            if name not in environ:
                break
```

Reply:
There was not an upper limit in mind.  It's harmless to have one, but there need to be a warning or error when the upper limit is hit.

Reply:
I updated the suggestion above to raise an error in case the name can't be found after checking `anonymous0-9` (you might want to tweak the error message/exception type).

Reply:
Side track: it's hard for me to decide whether or not to put a period at the end of an error message.  Sometimes I do sometimes I don't.  Any idea for a guideline?

Reply:
I tried a few expressions that raised errors and they don't seem to include the period.  Generally they are in the form of `ExceptionName: message`.

Reply:
Good to know that.  Thanks.  The for/else looks great too.

Reply:
I am separating this enhancement to a distinct PR: https://github.com/solvcon/modmesh/pull/153